HACKING.md

```diff
@@ -125,18 +127,94 @@ make
+$ gdb -q -p $(pidof hotspot) -ex "add-symbol-file hotspot-debuginfo-v1.4.0-43-g02fd82c/usr/bin/hotspot"
+Attaching to process 123456
+```
+
+with a GDB session of
+
+```txt
+Reading symbols from /tmp/.mount_hotspoBCRbef/squashfs-root/usr/bin/hotspot...
+
+(No debugging symbols found in /tmp/.mount_hotspoBCRbef/squashfs-root/usr/bin/hotspot)
+Reading symbols from /tmp/.mount_hotspoBCRbef/squashfs-root/usr/bin/../lib/libKF5ThreadWeaver.so.5...
+(No debugging symbols found in /tmp/.mount_hotspoBCRbef/squashfs-root/usr/bin/../lib/libKF5ThreadWeaver.so.5)
+...
+0x00007fea98ae0a38 in poll () from /usr/lib64/libc.so.6
+add symbol table from file "hotspot-debuginfo-v1.4.0-43-g02fd82c/usr/bin/hotspot"
+(y or n) y
+Reading symbols from hotspot-debuginfo-v1.4.0-43-g02fd82c/usr/bin/hotspot...
+(gdb) bt
+#0  0x00007ff72544ca38 in poll () from /usr/lib64/libc.so.6
+#1  0x00007ff7246777d7 in _xcb_in_read_block () from /usr/lib64/libxcb.so.1
+#2  0x00007ff7246754e9 in xcb_connect_to_fd () from /usr/lib64/libxcb.so.1
+#3  0x00007ff72467943c in xcb_connect_to_display_with_auth_info () from /usr/lib64/libxcb.so.1
+#4  0x00007ff7283a95ea in _XConnectXCB () from /usr/lib64/libX11.so.6
+#5  0x00007ff728399d89 in XOpenDisplay () from /usr/lib64/libX11.so.6
+#6  0x00007ff70f181527 in QXcbBasicConnection::QXcbBasicConnection(char const*) () from /tmp/squashfs-root/usr/plugins/platforms/../../lib/libQt5XcbQpa.so.5
+#7  0x00007ff70f15b072 in QXcbConnection::QXcbConnection(QXcbNativeInterface*, bool, unsigned int, char const*) ()
+   from /tmp/squashfs-root/usr/plugins/platforms/../../lib/libQt5XcbQpa.so.5
+#8  0x00007ff70f15e588 in QXcbIntegration::QXcbIntegration(QStringList const&, int&, char**) ()
+   from /tmp/squashfs-root/usr/plugins/platforms/../../lib/libQt5XcbQpa.so.5
+#9  0x00007ff7286ad41f in ?? () from /tmp/squashfs-root/usr/plugins/platforms/libqxcb.so
+#10 0x00007ff7267686cf in QGuiApplicationPrivate::createPlatformIntegration() () from /tmp/squashfs-root/usr/bin/../lib/libQt5Gui.so.5
+#11 0x00007ff726769c40 in QGuiApplicationPrivate::createEventDispatcher() () from /tmp/squashfs-root/usr/bin/../lib/libQt5Gui.so.5
+#12 0x00007ff72630c0d7 in QCoreApplicationPrivate::init() () from /tmp/squashfs-root/usr/bin/../lib/libQt5Core.so.5
+#13 0x00007ff72676c639 in QGuiApplicationPrivate::init() () from /tmp/squashfs-root/usr/bin/../lib/libQt5Gui.so.5
+#14 0x00007ff726e12a69 in QApplicationPrivate::init() () from /tmp/squashfs-root/usr/bin/../lib/libQt5Widgets.so.5
+#15 0x000000000043a929 in main (argc=<optimized out>, argv=<optimized out>) at /__w/hotspot/hotspot/src/main.cpp:69
+(gdb) continue
+
+```
+
+or with an extracted image:
+
+```bash
+$ ./hotspot-v1.4.0-43-g02fd82c-x86_64.AppImage --appimage-extract
+$ PATH=./squashfs-root/usr/bin:$PATH LD_LIBRARY_PATH=./squashfs-root/usr/lib64:/usr/local/lib64:/usr/lib64:./squashfs-root/usr/lib:$LD_LIBRARY_PATH gdb -q --exec=hotspot --symbols hotspot-debuginfo-v1.4.0-43-g02fd82c/usr/bin/hotspot -ex run
```

Comment:
maybe we should add some kind of `--appimage-debug gdb` or similar to our `AppRun` which would prepend the debugger before starting hotspot?

or maybe better: extract the code that sets up the env from the `AppRun`, such that we can do:

```
source appimage.env
gdb -q --exec=hotspot --symbols hotspot-debuginfo-v1.4.0-43-g02fd82c/usr/bin/hotspot -ex run
```

directly?

Reply:
appimage.env would be marvelous! Even better would be if appimage can be executed, too: `/path/to/appimage.env hotspot` (or, using gdb that way).

`--appimage-debug Name-Of-Debugger-Here [additional debugger arguments] [-- additional arguments passed to hotspot]` would be nice, too; allowing things like `/hotspot-v1.4.0-43-g02fd82c-x86_64.AppImage --appimage-debug /opt/path/to/newer/gdb/bin/gdb -x additional_printers.gdb --args -- some.perf.data`

Should we create one (or two) issue(s) for those good ideas?

Reply:
Should we create one (or two) issue(s) for those good ideas of yours?

Reply:
yes please do that